contracts/examples/seed-nft-minter/src/nft_module.rs

```diff
@@ -0,0 +1,205 @@
+        &self,
+        nft_nonce: u64,
+    ) -> OptionalValue<MultiValue3<EgldOrEsdtTokenIdentifier, u64, BigUint>> {
+        if self.price_tag(nft_nonce).is_empty() {
+            // NFT was already sold
+            OptionalValue::None
+        } else {
+            let price_tag = self.price_tag(nft_nonce).get();
+
+            OptionalValue::Some((price_tag.token, price_tag.nonce, price_tag.amount).into())
+        }
+    }
+
+    // callbacks
+
+    #[callback]
+    fn issue_callback(
+        &self,
+        #[call_result] result: ManagedAsyncCallResult<EgldOrEsdtTokenIdentifier>,
+    ) {
+        match result {
+            ManagedAsyncCallResult::Ok(token_id) => {
+                self.nft_token_id().set(&token_id.unwrap_esdt());
+            },
+            ManagedAsyncCallResult::Err(_) => {
+                let caller = self.blockchain().get_owner_address();
+                let returned = self.call_value().egld_or_single_esdt();
+                if returned.token_identifier.is_egld() && returned.amount > 0 {
+                    self.send()
+                        .direct(&caller, &returned.token_identifier, 0, &returned.amount);
+                }
+            },
+        }
+    }
+
+    // private
+
+    #[allow(clippy::too_many_arguments)]
+    fn create_nft_with_attributes<T: TopEncode>(
+        &self,
+        name: ManagedBuffer,
+        royalties: BigUint,
+        attributes: T,
+        uri: ManagedBuffer,
+        selling_price: BigUint,
+        token_used_as_payment: EgldOrEsdtTokenIdentifier,
+        token_used_as_payment_nonce: u64,
+    ) -> u64 {
+        self.require_token_issued();
+        require!(royalties <= ROYALTIES_MAX, "Royalties cannot exceed 100%");
+
+        let nft_token_id = self.nft_token_id().get();
+
+        let mut serialized_attributes = ManagedBuffer::new();
+        if let core::result::Result::Err(err) = attributes.top_encode(&mut serialized_attributes) {
+            sc_panic!("Attributes encode error: {}", err.message_bytes());
+        }
+
+        let attributes_sha256 = self.crypto().sha256(&serialized_attributes);
+        let attributes_hash = attributes_sha256.as_managed_buffer();
+        let uris = ManagedVec::from_single_item(uri);
+        let nft_nonce = self.send().esdt_nft_create(
+            &nft_token_id,
+            &BigUint::from(NFT_AMOUNT),
+            &name,
+            &royalties,
+            attributes_hash,
+            &attributes,
+            &uris,
+        );
+
+        self.price_tag(nft_nonce).set(&PriceTag {
+            token: token_used_as_payment,
+            nonce: token_used_as_payment_nonce,
+            amount: selling_price,
+        });
+
+        nft_nonce
+    }
+
+    fn require_token_issued(&self) {
+        require!(!self.nft_token_id().is_empty(), "Token not issued");
+    }
+
+    // storage
+
+    #[view(getNftTokenId)]
+    #[storage_mapper("nftTokenId")]
+    fn nft_token_id(&self) -> SingleValueMapper<TokenIdentifier>;
```

Comment:
I suggest using `NonFungibleTokenMapper`, would clear up a lot of the code from this file.

Reply:
Done.